src/futils.cpp

```diff
@@ -85,8 +85,10 @@ namespace Exiv2 {
             pstr++;
         }
         *pbuf = '\0';
-        return buf;
-    } // urlencode
+        std::string ret(buf);
+        free(buf);
```

Comment:
Good catch. Judging from your submitted diff, this was never freed' again. Congratulations for the fixed memory leak!

Could you change `malloc` & `free` to `new` & `delete[]`? The use of the C functions is discouraged in C++.

Reply:
It was not a memory leak, the code in `src/basicio.cpp` was releasing correctly the memory, but since Exiv2 is a C++ project I thought it does not make sense to leave the memory management responsibility in the client code.